### PR TITLE
[Type:Refactor] optimize some code in admin

### PR DESCRIPTION
**Pull request by li-keguo**
work list:

- [ ] permission used existed method.
- [ ] used method reference.
- [ ] switch case maybe is merged.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/AbstractDataChangedListener.java [hunk 1]
```diff
@@ -96,25 +95,13 @@
...
-                List<RuleData> ruleList = GsonUtils.getGson().fromJson(config.getJson(), new TypeToken<List<RuleData>>() {
-                }.getType());
-                return new ConfigData<>(config.getMd5(), config.getLastModifyTime(), ruleList);
             case SELECTOR:
-                List<SelectorData> selectorList = GsonUtils.getGson().fromJson(config.getJson(), new TypeToken<List<SelectorData>>() {
-                }.getType());
-                return new ConfigData<>(config.getMd5(), config.getLastModifyTime(), selectorList);
             case META_DATA:
-                List<MetaData> metaList = GsonUtils.getGson().fromJson(config.getJson(), new TypeToken<List<MetaData>>() {
+                List<?> appAuthList = GsonUtils.getGson().fromJson(config.getJson(), new TypeToken<List<?>>() {
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/AbstractDataChangedListener.java:102]: why refactor this？

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/AbstractDataChangedListener.java [hunk 2]
```diff
@@ -96,25 +95,13 @@
         ConfigDataCache config = CACHE.get(groupKey.name());
         switch (groupKey) {
             case APP_AUTH:
-                List<AppAuthData> appAuthList = GsonUtils.getGson().fromJson(config.getJson(), new TypeToken<List<AppAuthData>>() {
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/AbstractDataChangedListener.java:99]: why remove this？

Comment (li-keguo) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/AbstractDataChangedListener.java:99]: Maybe it can be merged

**Comment by li-keguo** [comment]
ConfigDataCache maybe save obj and json string, 
query is return obj[not convert],

**Review by yu199195 — APPROVED** [review]

**Pull request merged**